UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API/Services/FssEventProcessor.cs

```diff
@@ -0,0 +1,69 @@
+using UKHO.ExternalNotificationService.Common.Helpers;
+using UKHO.ExternalNotificationService.Common.Logging;
+using UKHO.ExternalNotificationService.Common.Models.EventModel;
+using UKHO.ExternalNotificationService.Common.Models.Request;
+using UKHO.ExternalNotificationService.Common.Models.Response;
+
+namespace UKHO.ExternalNotificationService.API.Services
+{
+    public class FssEventProcessor : IEventProcessor
+    {
+        private readonly IFssEventValidationAndMappingService _fssEventValidationAndMappingService;
+        private readonly ILogger<FssEventProcessor> _logger;
+        private List<Error> _errors;
+        private readonly IAzureEventGridDomainService _azureEventGridDomainService;
+
+        public string EventType => EventProcessorTypes.FSS;
+
+        public FssEventProcessor(IFssEventValidationAndMappingService fssEventValidationAndMappingService,
+                                 ILogger<FssEventProcessor> logger,
+                                 IAzureEventGridDomainService azureEventGridDomainService)
+        {
+            _fssEventValidationAndMappingService = fssEventValidationAndMappingService;
+            _logger = logger;
+            _azureEventGridDomainService = azureEventGridDomainService;
+        }
+
+        public async Task<ExternalNotificationServiceProcessResponse> Process(CustomEventGridEvent customEventGridEvent, string correlationId, CancellationToken cancellationToken = default)
+        {
+            string data =JsonConvert.SerializeObject(customEventGridEvent.Data);
+            FssEventData fssEventData = JsonConvert.DeserializeObject<FssEventData>(data);
+
+            ValidationResult validationFssEventData = await _fssEventValidationAndMappingService.ValidateFssEventData(fssEventData);
+
+            if (!validationFssEventData.IsValid && validationFssEventData.HasOkErrors(out _errors))
+                return ProcessResponse(fssEventData);
+
+            if (!string.IsNullOrWhiteSpace(BusinessUnitTypes.BusinessUnit.FirstOrDefault(x => x.Equals(fssEventData.BusinessUnit))))
+            {
+                _logger.LogInformation(EventIds.FssEventDataMappingStart.ToEventId(), "File share service event data mapping started for subject:{subject}, businessUnit:{businessUnit} and _X-Correlation-ID:{correlationId}.", customEventGridEvent.Subject, fssEventData.BusinessUnit, correlationId);
+                CloudEvent cloudEvent = _fssEventValidationAndMappingService.FssEventDataMapping(customEventGridEvent, correlationId);
+                _logger.LogInformation(EventIds.FssEventDataMappingCompleted.ToEventId(), "File share service event data mapping completed for subject:{subject}, businessUnit:{businessUnit} and _X-Correlation-ID:{correlationId}.", customEventGridEvent.Subject, fssEventData.BusinessUnit, correlationId);
+
+                await _azureEventGridDomainService.PublishEventAsync(cloudEvent, correlationId, cancellationToken);
```

Comment:
Like above, publishing to EG should be required by all processors and we should consider moving it to base class.

Reply:
I have made the required changes and moved it to base class.